torch/nn/modules/batchnorm.py

```diff
@@ -144,7 +144,7 @@ def forward(self, input: Tensor) -> Tensor:
         if self.training and self.track_running_stats:
             # TODO: if statement only here to tell the jit to skip emitting this when it is None
             if self.num_batches_tracked is not None:  # type: ignore[has-type]
-                self.num_batches_tracked = self.num_batches_tracked + 1  # type: ignore[has-type]
+                self.num_batches_tracked += 1  # type: ignore[has-type]
```

Comment:
Is the type ignore still necessary here?

Reply:
@albanD would I be correct that this is actually faster as we avoid one Tensor allocation / cache lookup? And this should still work even if DDP/SyncBatchNorm where the buffers are communicated through network, as those comm ops would synchronize the streams accordingly?

Reply:
> would I be correct that this is actually faster as we avoid one Tensor allocation / cache lookup?

I'm not sure where the cache lookup comes from? If from attribute access, then I guess we do just one read instead of a read and a set yes.
For the Tensor allocation, yes, but this is a size 1 Tensor so that should be pretty insignificant.

> And this should still work even if DDP/SyncBatchNorm where the buffers are communicated through network, as those comm ops would synchronize the streams accordingly?

I don't know how SyncBatchNorm work. and in particular how it detects these inplace changes and share them accross. So I can't really answer here.

Reply:
> I'm not sure where the cache lookup comes from?

Oh, I was referring to the CUDACachingAllocation lookup.

Reply:
> Is the type ignore still necessary here?

I must admit I don't know what that does (haven't done much Python lately). At some point mypy was telling me it was indeed unused (among other errors) but now that I've fixed the errors it doesn't say that anymore so... maybe it's still needed?